### ISSUE TITLE
Coverage options not caught and merged

### ISSUE THREAD

**Issue by Lakitna** [monounity/karma-typescript#290]
I'm running into an issue where I can't configure coverage as called by karma-typescript. After looking at documentation and in the source code I found out that the `coverageReporter` object should be merged with the karma-typescript specific defaults. It appears this doesn't happen for me.  
See https://github.com/monounity/karma-typescript/blob/master/src/shared/configuration.ts#L193 

The coverage settings do get applied when I add `coverage` to the reporters, but that also results in it running twice, once on the compiled files and once on the source files. The settings do get applied to the output of the compiled files. This tells me that at least my coverage settings are correct, but the script linked above doesn't work for some reason.

karma.conf.js:
```javascript
module.exports = function(config) {
  config.set({
    frameworks: [
        "mocha",
        "chai",
        "karma-typescript"
    ],
    reporters: [
        "progress",
        "karma-typescript"
    ],
    browsers: ["Chrome"],

    files: [
        { pattern: "./base.spec.ts" },
        { pattern: "src/**/*.+(ts|html)" }
    ],

    preprocessors: {
        "**/*.ts": ["karma-typescript"]
    },

    coverageReporter: {
        dir : 'coverage/',
        reporters: [
            { type: 'html', subdir: '.'},
            { type: 'text-summary' }
        ]
    },

    karmaTypescriptConfig: {
        bundlerOptions: {
            entrypoints: /\.spec\.ts$/,
            transforms: [
                require("karma-typescript-angular2-transform")
            ]
        },
        compilerOptions: {
            lib: ["ES2015", "DOM"]
        }
    },
  });
};
```

**Comment by Lakitna** [comment]
So I've done a bit more investigating. 

The file I linked to does successfully combine the default settings and the custom ones from karma.conf.js. It also successfully adds `coverage` to the `reporters` array when required.

It seems like even though the `converageReporter` settings are successfully extracted from karma.conf.js, they aren't actually applied when running the coverage reporter. I haven't been able to trace where it goes wrong.

Is there anyone who can tell me where the combined `converageReporter` settings are actually applied?

---

The `reporters` array is a different story. The behaviour I'm seeing is that coverage will run twice when I add `coverage` to the `reporters` array in karma.conf.js. Once with the karma-typescript default config on the source .ts files, and once with my custom config on the compiled .ts files. When I don't add `coverage` to the `reporters` array it only runs once with the karma-typescript defaults on the source .ts files.

This tells me that karma-typescript runs coverage regardless of the contents of karma.conf.js and always with the default settings.

**Comment by Lakitna** [comment]
# Fixed it

So I've found out that my karma.conf.js was the problem. I should have used the `karmaTypescriptConfig` to configure the coverage reporters and stuff.

I also realised that it all went wrong in the documentation. When I wanted to change the coverage reporter settings I went to the repo root and searched for the text `coverage` to see how I should set things. The first hit is the following code snippet:

```javascript
coverageReporter: {
    instrumenterOptions: {
        istanbul: { noCompact: true }
    }
}
```

I recognised that structure from karma-coverage so I used their way of settings things. It took me a long time to find out this was wrong and this could have been prevented by one of two things:

1. Actually use the same settings structure as karma-coverage for this part
2. Make the actual way of configuring easier to find in the documentation

I'd prefer the first solution.

**Comment by erikbarke** [comment]
Thanks for reporting, I'll see what I can do about this!

**Comment by erikbarke** [comment]
Fixed in #302.
